akka-cluster-sharding/src/test/scala/akka/cluster/sharding/StopShardsSpec.scala

```diff
@@ -0,0 +1,133 @@
+    ConfigFactory.parseString("""
+        akka.loglevel = DEBUG
+        akka.loggers = ["akka.testkit.SilenceAllTestEventListener"]
+        akka.actor.provider = "cluster"
+        akka.remote.artery.canonical.port = 0
+        akka.test.single-expect-default = 5 s
+        akka.cluster.sharding.distributed-data.durable.keys = []
+        akka.cluster.sharding.remember-entities = off
+        akka.cluster.downing-provider-class = akka.cluster.testkit.AutoDowning
+        akka.cluster.jmx.enabled = off
+        akka.cluster.sharding.verbose-debug-logging = on
+        akka.cluster.sharding.fail-on-invalid-entity-state-transition = on
+        akka.remote.artery.canonical.hostname = "127.0.0.1"
+        """)
+
+  val shardTypeName = "stopping-entities"
+
+  val numberOfShards = 3
+
+  val extractEntityId: ShardRegion.ExtractEntityId = {
+    case msg: Int => (msg.toString, msg)
+    case _        => throw new IllegalArgumentException()
+  }
+
+  val extractShardId: ShardRegion.ExtractShardId = {
+    case msg: Int                    => (msg % 10).toString
+    case ShardRegion.StartEntity(id) => (id.toLong % numberOfShards).toString
+    case _                           => throw new IllegalArgumentException()
+  }
+
+  case class Pong(actorRef: ActorRef)
+  class EntityActor extends Actor with ActorLogging {
+    override def receive: Receive = {
+      case _ =>
+        log.debug("ping")
+        sender() ! context.self
+    }
+  }
+}
+
+class StopShardsSpec extends AkkaSpec(StopShardsSpec.config) with WithLogCapturing {
+  import StopShardsSpec._
+
+  // mute logging of deadLetters
+  system.eventStream.publish(Mute(DeadLettersFilter[Any]))
+
+  private val sysA = system
+  private val sysB = ActorSystem(system.name, system.settings.config)
+
+  private val pA = TestProbe()(sysA)
+  private val pB = TestProbe()(sysB)
+
+  private val regionA = startShardregion(sysA)
+  private val regionB = startShardregion(sysB)
+
+  "The StopShards command" must {
+
+    "form a cluster" in {
+      Cluster(sysA).join(Cluster(sysA).selfAddress) // coordinator on A
+      awaitAssert(Cluster(sysA).selfMember.status shouldEqual MemberStatus.Up, 3.seconds)
+
+      Cluster(sysB).join(Cluster(sysA).selfAddress)
+      awaitAssert(Cluster(sysB).selfMember.status shouldEqual MemberStatus.Up, 3.seconds)
+
+      // wait for all regions to be registered
+      // FIXME test depends on this right now, stopping should work also when if rebalancing because of topology change?
+      pA.awaitAssert({
+        regionA.tell(GetCurrentRegions, pA.ref)
+        pA.expectMsgType[CurrentRegions].regions should have size (2)
+      }, 10.seconds)
+    }
+
+    "start entities in a few shards, then stop the shards" in {
+
+      val allShards = (1 to 10)
+        .map { i =>
+          regionA.tell(i, pA.ref)
+          val entityRef = pA.expectMsgType[ActorRef]
+          pA.watch(entityRef) // so we can verify terminated later
+          extractShardId(i) -> entityRef
```

Comment:
the entityRef isn't used, so this could be just the shard and remove next `.map`?